File: RanKing-app/backend/app/schemas/contest.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class ContestCreate(BaseModel):
    name: str
    description: Optional[str] = None
    submission_start_date: datetime
    submission_end_date: datetime
    voting_end_date: datetime


class ContestResponse(BaseModel):
    contest_id: int
    name: str
    description: Optional[str] = None
    submission_start_date: datetime
    submission_end_date: datetime
    voting_end_date: datetime

    class Config:
        from_attributes = True

File: RanKing-app/ios-client/RanKing/RanKing/Models/Contest.swift
//
//  Contest.swift
//  RanKing
//
//  Created by Jonathan Da Silva on 10/12/25.
//

import Foundation
import SwiftData

final class ContestResponse : Codable  {
    var statusCode: Int
    var body: String
}

final class Contest : Codable  {
    var contest_id : String
    var name: String
    var description: String
    var submission_start_date: Date
    var submission_end_date: Date
    var voting_end_date : Date
}
